src/Avalonia.Controls/Primitives/UniformGrid.cs

```diff
@@ -116,6 +116,22 @@ protected override Size MeasureOverride(Size availableSize)
                 }
             }
 
+            // If layout rounding is enabled, round the measured per-cell unit size (maxWidth/maxHeight).
+            // We round the measured max sizes instead of the childAvailableSize to avoid changing the input
+            // used for children.Measure(...) which could change how children measure themselves.
+            if (UseLayoutRounding)
+            {
+                if (!double.IsInfinity(maxWidth) && !double.IsNaN(maxWidth))
```

Comment:
Use `!MathUtilities.IsFinite()` instead (I'm leaving this once, but this applies to the 4 added `if` conditions).